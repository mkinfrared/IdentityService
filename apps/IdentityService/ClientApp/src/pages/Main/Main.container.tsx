import { memo } from "react";
import { useLocation } from "react-use";

import getRoutePath from "utils/getRoutePath";

import MainFC from "./Main";
import { MainProps } from "./Main.type";

const Main = (props: Omit<MainProps, "path">) => {
  const { pathname } = useLocation();

  if (!pathname) {
    return null;
  }

  const path = getRoutePath(pathname);

  return <MainFC {...props} path={path} />;
};

export default memo(Main);
