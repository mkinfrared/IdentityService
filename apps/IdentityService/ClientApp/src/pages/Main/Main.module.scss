@import "../../styles/common";

.Main {
  @extend %loginRegisterPage;

  .formContainer {
    justify-content: flex-start;
  }
}

.Story {
}
