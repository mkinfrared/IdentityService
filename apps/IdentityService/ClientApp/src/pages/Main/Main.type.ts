export type MainProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  path: Routes;
};

export enum Routes {
  CONSENT = "consent",
  CONFIRM_EMAIL = "confirmEmail",
  LOGIN = "login",
  REGISTER = "register",
  FORGOT_PASSWORD = "forgotPassword",
  PASSWORD_RESET = "resetPassword",
}
