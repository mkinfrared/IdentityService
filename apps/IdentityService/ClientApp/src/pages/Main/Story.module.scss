.Story {
  .buttonContainer {
    background-color: palegoldenrod;
    padding: 1rem;

    .storyButton {
      margin-right: 1rem;
    }
  }
}
