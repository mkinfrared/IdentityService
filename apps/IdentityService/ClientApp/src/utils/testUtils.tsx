import { ReactNode } from "react";
import { QueryClient, QueryClientProvider } from "react-query";

type WithQueryProps = {
  children?: ReactNode;
};

const withQuery = ({ children }: WithQueryProps) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
};

export { withQuery };
