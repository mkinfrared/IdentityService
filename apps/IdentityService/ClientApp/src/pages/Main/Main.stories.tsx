import { Meta, Story } from "@storybook/react";

import { Main } from "./Main";
import css from "./Main.module.scss";
import { MainProps, Routes } from "./Main.type";

export default {
  title: "PAGES/Main",
  component: Main,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<MainProps> = (args) => (
  <div className={css.Story}>
    <Main {...args} />
  </div>
);

const Default = Template.bind({});

Default.args = {
  path: Routes.LOGIN,
};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

export { Default };
