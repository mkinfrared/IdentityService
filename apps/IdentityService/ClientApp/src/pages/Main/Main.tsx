import { classNames, lazyImport } from "@identity-service/core";
import { Card, Spinner } from "@identity-service/ui";
import { Suspense, memo } from "react";

import { useReturnUrl } from "hooks";

import css from "./Main.module.scss";
import { MainProps } from "./Main.type";

const ConsentForm = lazyImport(
  () =>
    import(
      /* webpackChunkName: "ConfirmEmail" */
      /* webpackPrefetch: true */
      "components/ConsentForm"
    ),
);

const ConfirmEmail = lazyImport(
  () =>
    import(
      /* webpackChunkName: "ConfirmEmail" */
      /* webpackPrefetch: true */
      "components/ConfirmEmailForm"
    ),
);

const LoginForm = lazyImport(
  () =>
    import(
      /* webpackChunkName: "LoginForm" */
      /* webpackPrefetch: true */
      "components/LoginForm"
    ),
);

const RegisterForm = lazyImport(
  () =>
    import(
      /* webpackChunkName: "RegisterForm" */
      /* webpackPrefetch: true */
      "components/RegisterForm"
    ),
);

const Main = ({ className, path }: MainProps) => {
  const returnUrl = useReturnUrl();

  let child: JSX.Element;

  switch (path) {
    case "login":
      child = <LoginForm className={css.formContainer} returnUrl={returnUrl} />;

      break;
    case "register":
      child = (
        <RegisterForm className={css.formContainer} returnUrl={returnUrl} />
      );

      break;
    case "confirmEmail":
      child = (
        <ConfirmEmail className={css.formContainer} returnUrl={returnUrl} />
      );

      break;

    case "consent":
      child = (
        <ConsentForm className={css.formContainer} returnUrl={returnUrl} />
      );

      break;

    default:
      child = <LoginForm className={css.formContainer} returnUrl={returnUrl} />;
  }

  return (
    <div className={classNames(css.Main, className)}>
      <Card className={css.card}>
        <Suspense
          fallback={<Spinner className={css.spinner} type="infinity" />}
        >
          {child}
        </Suspense>
      </Card>
    </div>
  );
};

export { Main };

export default memo(Main);

/* webpackInclude: /\.json$/ */
/* webpackExclude: /\.noimport\.json$/ */
/* webpackChunkName: "my-chunk-name" */
/* webpackMode: "lazy" */
/* webpackPrefetch: true */
/* webpackPreload: true */
