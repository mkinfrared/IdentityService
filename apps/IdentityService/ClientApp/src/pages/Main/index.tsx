import "@identity-service/ui/lib/esm/styles.min.css";
import "index.scss";
import ReactDOM from "react-dom";

import Main from "./Main.container";

ReactDOM.render(<Main />, document.getElementById("app"));
