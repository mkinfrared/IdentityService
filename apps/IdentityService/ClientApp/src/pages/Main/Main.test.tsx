import { render, waitFor } from "@testing-library/react";

import { Routes } from "pages/Main/Main.type";
import { withQuery } from "utils/testUtils";

import { Main } from "./Main";

jest.mock("components/ConfirmEmailForm");

jest.mock("components/ForgotPasswordForm");

jest.mock("components/LoginForm");

jest.mock("components/RegisterForm");

jest.mock("components/PasswordResetForm");

describe("<Main />", () => {
  it("should match the snapshot", () => {
    const { container } = render(<Main path={Routes.LOGIN} />);

    expect(container).toMatchSnapshot();
  });

  it("should match the snapshot", () => {
    const { container } = render(<Main path={Routes.LOGIN} />);

    expect(container).toMatchSnapshot();
  });

  it("should render <LoginForm /> when path is 'login'", async () => {
    const testId = "LoginForm";
    const { findByTestId } = render(<Main path={Routes.LOGIN} />);

    await waitFor(async () => {
      const element = await findByTestId(testId);

      expect(element).toBeDefined();
    });
  });

  it("should render <RegisterForm /> when path is 'register'", async () => {
    const testId = "RegisterForm";
    const { findByTestId } = render(<Main path={Routes.REGISTER} />);

    await waitFor(async () => {
      const element = await findByTestId(testId);

      expect(element).toBeDefined();
    });
  });

  it("should render <ConfirmEmailForm /> when path is 'confirmEmail'", async () => {
    const testId = "ConfirmEmailForm";
    const { findByTestId } = render(<Main path={Routes.CONFIRM_EMAIL} />);

    await waitFor(async () => {
      const element = await findByTestId(testId);

      expect(element).toBeDefined();
    });
  });

  it("should render <ConsentForm /> when path is 'consent'", async () => {
    const testId = "ConsentForm";

    const { findByTestId } = render(<Main path={Routes.CONSENT} />, {
      wrapper: withQuery,
    });

    await waitFor(async () => {
      const element = await findByTestId(testId);

      expect(element).toBeDefined();
    });
  });

  it("should render <ForgotPasswordForm /> when path is 'forgotPassword'", async () => {
    const testId = "ForgotPasswordForm";

    const { findByTestId } = render(<Main path={Routes.FORGOT_PASSWORD} />, {
      wrapper: withQuery,
    });

    await waitFor(async () => {
      const element = await findByTestId(testId);

      expect(element).toBeDefined();
    });
  });

  it("should render <PasswordResetForm /> when path is 'resetPassword'", async () => {
    const testId = "PasswordResetForm";

    const { findByTestId } = render(<Main path={Routes.PASSWORD_RESET} />, {
      wrapper: withQuery,
    });

    await waitFor(async () => {
      const element = await findByTestId(testId);

      expect(element).toBeDefined();
    });
  });

  it("should render <LoginForm /> as the default value", async () => {
    const testId = "LoginForm";
    const { findByTestId } = render(<Main path={"marklar" as any} />);

    await waitFor(async () => {
      const element = await findByTestId(testId);

      expect(element).toBeDefined();
    });
  });
});
