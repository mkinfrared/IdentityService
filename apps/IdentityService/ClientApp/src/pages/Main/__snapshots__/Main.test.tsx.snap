// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should match the snapshot 1`] = `
<div>
  <div
    class="Main"
  >
    <div
      data-testid="LoginForm"
    >
      {"className":"formContainer","returnUrl":""}
    </div>
  </div>
</div>
`;

exports[`<Main /> should match the snapshot 2`] = `
<div>
  <div
    class="Main"
  >
    <div
      data-testid="LoginForm"
    >
      {"className":"formContainer","returnUrl":""}
    </div>
  </div>
</div>
`;
